docs/reference/search/search-your-data/search-with-synonyms.asciidoc

```diff
@@ -0,0 +1,138 @@
+
+An example synonyms file:
+
+[source,synonyms]
+--------------------------------------------------
+include::{es-test-dir}/cluster/config/analysis/synonym.txt[]
+--------------------------------------------------
+
+To update an existing synonyms set, upload new files to your cluster.
+Synonyms set files must be kept in sync on every cluster node.
+
+When a synonyms set is updated, search analyzers that use it need to be refreshed using the <<indices-reload-analyzers,reload search analyzers API>>
+
+This manual syncing and reloading makes this approach less flexible than using the <<synonyms-store-synonyms-api,synonyms API>>.
+
+[discrete]
+[[synonyms-store-synonyms-inline]]
+==== Inline
+
+You can test your synonyms by adding them directly inline in your token filter definition.
+
+[WARNING]
+======
+Inline synonyms are not recommended for production usage.
+A large number of inline synonyms increases cluster size unnecessarily and can lead to performance issues.
+======
+
+[discrete]
+[[synonyms-synonym-token-filters]]
+=== Configure synonyms token filters and analyzers
+
+Once your synonyms sets are created, you can start configuring your token filters and analyzers to use them.
+
+{es} uses synonyms as part of the <<analysis-overview,analysis process>>.
+You can use two types of <<analysis-tokenfilters,token filter>> to include synonyms:
+
+* <<analysis-synonym-graph-tokenfilter>>: It is recommended to use it, as it can correctly handle multi-word synonyms ("hurriedly", "in a hurry").
+* <<analysis-synonym-tokenfilter>>: Not recommended if you need to use multi-word synonyms.
+
+Check each synonym token filter documentation for configuration details and instructions on adding it to an analyzer.
+
+[discrete]
+[[synonyms-test-analyzer]]
+==== Test your analyzer
+
+You can test an analyzer configuration without modifying your index settings.
+Use the <<indices-analyze,analyze API>> to test your analyzer chain:
+
+[source,console]
+----
+GET /_analyze
+{
+  "tokenizer": "standard",
+  "filter" : [
+    "lowercase",
+    {
+      "type": "synonym_graph",
+      "synonyms": ["pc => personal computer", "computer, pc, laptop"]
+    }
+  ],
+  "text" : "Check how PC synonyms work"
+}
+----
+
+[discrete]
+[[synonyms-apply-synonyms]]
+==== Apply synonyms at index or search time
+
+Analyzers can be applied at <<analysis-index-search-time,index time or search time>>.
+
+You need to decide when to apply your synonyms:
+
+* Index time: Synonyms are applied when the documents are indexed into Elasticsearch. This is a less flexible alternative, as changes to your synonyms require <<docs-reindex,reindexing>>.
+* Search time: Synonyms are applied when a search is executed. This is a more flexible approach, which doesn't require reindexing. With this approach, search analyzers can be <<indices-reload-analyzers,reloaded>> whenever you make changes to your synonyms.
+
```

Comment:
About reload, let's add a clarification. Not all search time analyzers can be reloaded.  Let's modify to something like this:

Search time: Synonyms are applied when a search is executed. This is a more flexible approach, which doesn't require reindexing  when you need to introduce changes to synonyms. With this approach, if you set your synonym token filter to updateable:true,  it can be <<indices-reload-analyzers,reloaded>> whenever you make changes to your synonyms.